fem/tmop.hpp

```diff
@@ -804,6 +804,8 @@ class AdaptivityEvaluator
 
    virtual void ComputeAtNewPosition(const Vector &new_nodes,
                                      Vector &new_field) = 0;
+
+   void ClearGeometricFactors();
```

Comment:
Do you have to call this between remesh steps? Was it done at another place before?

Reply:
> Do you have to call this between remesh steps?

Yeah, right after each one


> Was it done at another place before?
 
It happened at the beginning of the remesh step (so it lived much longer), here:

https://github.com/mfem/mfem/blob/master/fem/tmop_tools.cpp#L258-L261

If you're interested I have the stack trace for that ^

Reply:
This method seems unnecessary -- you can always simply call `mesh->DeleteGeometricFactors()` -- no need to create a new method here for just that.

Reply:
I was hoping to do that and I may be missing something but since the mesh is constructed and owned by this class I didn't see a way to get at the object:

https://github.com/mfem/mfem/blob/master/fem/tmop.cpp#L2165-L2166

Although looking closer I see it's copy-constructed so maybe a copy-constructed mesh shared geo facts with its parent?

Reply:
Hmm, this is strange -- I don't see a compelling reason to copy the `Mesh`/`ParMesh`. @vladotomov, is this copy really necessary? You can always temporarily `SwapNodes` if you just need to assemble at different node positions.

Reply:
No, the copy is not really necessary. I copied it to simplify the logic, as I didn't want to iterate remaps on the same mesh nodes that are being optimized. We can optimize this and do it on the same `Mesh` object with `SwapNodes`, but it must be done carefully.

Do you think it's urgent to do it now or can it be done later?

Reply:
I don't see it as something urgent -- I was just surprised to see these mesh copies.